### ISSUE TITLE
Inlinining failed

### ISSUE THREAD

**Issue by feliwir** [g-truc/glm#512]
Hey i get the following error when using glm with the GLM_FORCE_AVX macro set:

`/usr/lib/gcc/x86_64-linux-gnu/5/include/popcntintrin.h: In function ‘int glm::bitCount(genType) [with genType = unsigned int]’:
/usr/lib/gcc/x86_64-linux-gnu/5/include/popcntintrin.h:35:1: error: inlining failed in call to always_inline ‘int _mm_popcnt_u32(unsigned int)’: target specific option mismatch
 _mm_popcnt_u32 (unsigned int __X)
 ^`

I also get this error (or similiar errors) for any other SIMD options.

**Comment by Groovounet** [comment]
Hi,

I couldn't reproduce this issue using GCC 4.6 or 4.9. 
Could you compile with #define GLM_MESSAGE before your first include and report the log here?

It will return something like the following which would help me to reproduce the issue:
```#pragma message: GLM: version 0.9.8.0
#pragma message: GLM: GCC compiler detected
#pragma message: GLM: 64 bits model
#pragma message: GLM: AVX2 instruction set
#pragma message: GLM: C++11
#pragma message: GLM: Swizzling operators disabled, #define GLM_SWIZZLE to enable swizzle operators
#pragma message("GLM: .length() returns glm::length_t, a typedef of int following the GLSL specification"```

Thanks,
Christophe

**Comment by feliwir** [comment]
I have no idea why, but the error doesn't occur anymore now

**Comment by Groovounet** [comment]
It's possible I fixed it on the way!

Thanks!
Christophe

**Comment by visorz** [comment]
I get the same errors, but only in DEBUG. My System:

Apple LLVM version 7.3.0 (clang-703.0.31)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
CPU is Intel i7-3667U with AVX according to [Intel ARK](https://ark.intel.com/products/64898/Intel-Core-i7-3667U-Processor-4M-Cache-up-to-3_20-GHz)

Messages in DEBUG:
GLM: version 0.9.8.3
GLM: Clang compiler detected
GLM: 64 bits model
GLM: AVX instruction set
GLM: C++11
GLM: Swizzling operators disabled, #define GLM_SWIZZLE to enable swizzle operators
GLM: Depth clip space: negative one to one
GLM: Coordinate system: right handed
GLM: .length() returns glm::length_t, a typedef of int following the GLSL specification

Here the errors:
In file included from /usr/local/include/glm/glm.hpp:88:
In file included from /usr/local/include/glm/integer.hpp:6:
In file included from /usr/local/include/glm/detail/func_integer.hpp:203:
In file included from /usr/local/include/glm/detail/func_integer.inl:366:
/usr/local/include/glm/detail/func_integer_simd.inl:54:10: error: use of undeclared identifier '_mm_popcnt_u32'
                return _mm_popcnt_u32(x);
                       ^
/usr/local/include/glm/detail/func_integer_simd.inl:61:27: error: use of undeclared identifier '_mm_popcnt_u64'
                return static_cast<int>(_mm_popcnt_u64(x));
                                        ^
In RELEASE mode no errors are emitted.

**Comment by Groovounet** [comment]
I think this issue is fixed in GLM 0.9.8 and master branch but it would be great if you could double check.

Thanks for reporting,
Christophe
